tests/PHPStan/Rules/DeadCode/data/bug-8620.php

```diff
@@ -0,0 +1,33 @@
+<?php declare(strict_types = 1);
+
+namespace Bug8620;
+
+use function PHPStan\Testing\assertType;
+
+class HelloWorld
+{
+	public function nullCoalesceAndConcatenation (?int $a = null): int
+	{
+		$key = ($a ?? "x") . "-";
+		assertType('non-falsy-string', $key);
```

Comment:
You still need to reference this file in NodeScopeResolverTest in order for these asserts to fail.